system-contract-dapp-playground/src/api/hedera/tokenManagement-interactions/index.ts

```diff
@@ -0,0 +1,536 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {
+  CommonKeyObject,
+  IHederaTokenServiceExpiry,
+  IHederaTokenServiceHederaToken,
+  TokenManagementSmartContractResult,
+} from '@/types/contract-interactions/HTS';
+import { prepareHederaTokenKeyArray } from '@/utils/contract-interactions/HTS/helpers';
+import { Contract, isAddress } from 'ethers';
+
+/**
+ * @dev manages and updates token information
+ *
+ * @dev integrates tokenMagemnentContract.updateTokenInfoPublic()
+ *
+ * @dev integrates tokenMagemnentContract.updateTokenExpiryInfoPublic()
+ *
+ * @dev integrates tokenMagemnentContract.updateTokenKeysPublic()
+ *
+ * @param baseContract: ethers.Contract
+ *
+ * @param API: "UPDATE_INFO" | "UPDATE_EXPIRY" | "UPDATE_KEYS"
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @param tokenInfo?: IHederaTokenServiceHederaToken
+ *
+ * @param expiryInfo?: IHederaTokenServiceExpiry
+ *
+ * @param keysInfo?: CommonKeyObject[],
+ *
+ * @return Promise<TokenManagementSmartContractResult>
+ */
+export const manageTokenInfomation = async (
+  baseContract: Contract,
+  API: 'UPDATE_INFO' | 'UPDATE_EXPIRY' | 'UPDATE_KEYS',
+  hederaTokenAddress: string,
+  tokenInfo?: IHederaTokenServiceHederaToken,
+  expiryInfo?: IHederaTokenServiceExpiry,
+  keysInfo?: CommonKeyObject[]
+): Promise<TokenManagementSmartContractResult> => {
+  // sanitize param
+  if (!isAddress(hederaTokenAddress)) {
+    console.error('Invalid token address');
+    return { err: 'Invalid token address' };
+  }
+
+  // invoking contract methods
+  try {
+    // prepare states
+    let transactionResult, errMsg;
+    switch (API) {
+      case 'UPDATE_INFO':
+        if (!tokenInfo) {
+          errMsg = 'Token information object is needed for UPDATE_INFO API';
+        } else {
+          transactionResult = await baseContract.updateTokenInfoPublic(
+            hederaTokenAddress,
+            tokenInfo
+          );
+        }
+        break;
+      case 'UPDATE_EXPIRY':
+        if (!expiryInfo) {
+          errMsg = 'Expiry information object is needed for UPDATE_EXPIRY API';
+        } else {
+          transactionResult = await baseContract.updateTokenExpiryInfoPublic(
+            hederaTokenAddress,
+            expiryInfo
+          );
+        }
+        break;
+      case 'UPDATE_KEYS':
+        if (!keysInfo) {
+          errMsg = 'Keys information object is needed for UPDATE_KEYS API';
+        } else {
+          // prepare keys array
+          const keyRes = prepareHederaTokenKeyArray(keysInfo);
+
+          // handle error
+          if (keyRes.err) {
+            errMsg = keyRes.err;
+          } else {
+            const hederaTokenKeys = keyRes.hederaTokenKeys;
+
+            transactionResult = await baseContract.updateTokenKeysPublic(
+              hederaTokenAddress,
+              hederaTokenKeys
+            );
+          }
+        }
+    }
+
+    // return err if any
```

Comment:
From here till the return is repeated logic across the methods.
Just move to a shared method and pass in the `errMsg`, `transactionResult`
```suggestion
    return handleResponse(transactionResult, errMsg);
```

```ts
const handleResponse = (transactionResult, errMsg) => {
    if (errMsg) {
      console.error(errMsg);
      return { err: errMsg };
    } else if (!transactionResult) {
      console.error('Cannot execute contract methods');
      return { err: 'Cannot execute contract methods' };
    }

    // get transaction receipt
    const txReceipt = await transactionResult.wait();

    // retrieve responseCode from event
    const { data } = txReceipt.logs.filter(
      (event: any) => event.fragment.name === 'ResponseCode'
    )[0];

    // @notice: 22 represents the predefined response code from the Hedera system contracts, indicating a successful transaction.
    return { result: Number(data) === 22, transactionHash: txReceipt.hash };
} 
```